tao_compiler/mlir/disc/transforms/fusion_utils.cc

```diff
@@ -2521,5 +2822,1033 @@ bool StitchCPUAnalysis::doCodeGeneration(OpBuilder& b, lmhlo::FusionOp fusion) {
   return true;
 }
 
+////////////////////// Stitch GPU FusionStrategy Implemenation /////////
+////////////////////////////////////////////////////////////////////////
+
+Value StitchGpuFusionStrategy::getEffectiveShape(FusionPattern& target,
+                                                 Value v) {
+  Operation* result_op = target.findLastWriter(v);
+  assert(result_op);
+  // effective shape of reduce op is its operand's shape.
+  return isa<lmhlo::ReduceOp>(result_op) ? result_op->getOperand(0) : v;
+}
+
+bool StitchGpuFusionStrategy::tileInfoPropagateI2O(
```

Comment:
This function is not used anywhere. Is this within expectation?

Reply:
It is indeed not used in the final implementation. I leave it here incase we need it someday.